src/fw_update.c

```diff
@@ -257,14 +301,25 @@ static void fw_update_thread(void *arg)
         download_ctx.sha = golioth_sys_sha256_create();
 
         int err = GOLIOTH_ERR_FAIL;
-        while (err)
+        while (1)
         {
             err = golioth_ota_download_component(_client,
                                                  _main_component,
                                                  &next_block,
                                                  fw_write_block_cb,
                                                  (void *) &download_ctx);
-            if (err != GOLIOTH_OK)
+
+            if (err == GOLIOTH_ERR_IO)
+            {
+                GLTH_LOGE(TAG, "Failed to store OTA component");
+                break;
+            }
+
+            if (err == GOLIOTH_OK)
+            {
+                break;
+            }
+            else
```

Comment:
Should we include a delay here?

Reply:
We could. Is there a reason to include a delay?

At the top of the loop we will add a new request to the mailbox so this will get put at the end of the line if there are any other network operations in progress. If the client is not running this will just queue the operation until the client starts again.

Reply:
If there are error conditions that cause the request to fail immediately (like the mailbox being full), this will be a tight loop and quickly exhaust the retries. Even if there is nothing else going on, whatever the transient reason that caused the firmware update to fail could exist for some period of time. I think it's useful here to have a small delay that gives whatever error condition time to pass. What do you think?

Reply:
I see, that's a good point.

I added a delay of 15 seconds. When combined with 15 max resumes on a single block we have a delay of up to 3:45 which should be ample for a cellular connection to be re-established before the download is considered a failure.

Reply:
I'm starting to feel like I can guess your favorite number 😉